src/libs/antares/study/parts/hydro/series.h

```diff
@@ -105,6 +105,8 @@ class DataSeriesHydro
     */
     void checkMinGenTsNumber(Study& s, const AreaName& areaID);
 
+    unsigned int getIndex(unsigned int year) const;
```

Comment:
I think that every time we get this index, it is only to use it later to get one of the 3 corresponding columns in ROR / storage / mingen.

I would be in favor, instead, of returning straight from here what is actually useful, which are those 3 columns, in a small struct:
```cpp
struct YearlyHydroSeries {
    Matrix<double, int32_t>::ColumnType ror;
    Matrix<double, int32_t>::ColumnType storage;
    Matrix<double, int32_t>::ColumnType mingen;
};

class DataSeriesHydro {
...
    YearlyHydroSeries getSeries(unsigned int year);
...
};
```
or possibly 3 getters to get the columns.

But maybe for a later PR, since future refactorings are planned ?

Reply:
See #1669 ;)